.cicdtemplate/.codemagic/codemagic.yaml

```diff
@@ -97,3 +99,67 @@ workflows:
       - *unit_test_on_template_xml
     artifacts:
       - *artifacts_template_xml
+
+  build-and-deploy-template-compose-staging:
+    name: Build and deploy template-compose staging to Firebase App Distribution
+    <<: *configure_environment
+    environment:
+      groups:
+        - firebase_credentials
+        - build_configure
+    triggering:
+      events:
+        - push
+      branch_patterns:
+        - pattern: develop
+        - pattern: main
+    scripts:
+      - *detekt_on_template_compose
+      - *unit_test_on_template_compose
+      - name: Build APK for staging
+        working_directory: ./template-compose
+        script: |
+          ./gradlew assembleStagingDebug -PversionCode=$BUILD_NUMBER
+    artifacts:
+      - *artifacts_template_compose
+      - *artifacts_staging_apk
+    publishing:
+      firebase:
+        firebase_service_account: $FIREBASE_SERVICE_ACCOUNT_CREDENTIALS
+        android:
+          app_id: 1:215362241811:android:51e902760fcff1243446e6
+          groups:
+            - android-chapter
+          artifact_type: 'apk'
+
+  build-and-deploy-template-compose-production:
+    name: Build and deploy template-compose production to Firebase App Distribution
+    <<: *configure_environment
+    environment:
+      groups:
+        - firebase_credentials
+        - build_configure
+    triggering:
+      events:
+        - push
+      branch_patterns:
+        - pattern: develop
+        - pattern: main
+    scripts:
+      - *detekt_on_template_compose
+      - *unit_test_on_template_compose
+      - name: Build APK for production
+        working_directory: ./template-compose
+        script: |
+          ./gradlew assembleProductionDebug -PversionCode=$BUILD_NUMBER
```

Comment:
Should we make a release build for production falvor to have a chance to verify the app release-readiness? This requires us to create a keystore and save it in 1Pass 🙏

Reply:
@luongvo I think we should do the same we did with the [Github action here](https://github.com/nimblehq/android-templates/blob/develop/.cicdtemplate/.github/workflows/deploy_staging_and_production_to_firebase_app_distribution.yml#L73)

Reply:
@luongvo Let me split this change once we update the Github action script. 🙏🏻

Reply:
@lydiasama Do we have a note on an existing task already for me to resolve this comment?

Reply:
@luongvo Not yet. I'm about to create the issue. 🙏🏻

Reply:
The issue is created [here](https://github.com/nimblehq/android-templates/issues/491). 🙇🏻‍♀️